cellpack/autopack/Analysis.py

```diff
@@ -927,6 +1010,335 @@ def PairCorrelationFunction_2D(self, x, y, S, rMax, dr):
+            )
+
+    def get_obj_dict(self, input_path):
+        """
+        Returns the object dictionary from the input path folder.
+        TODO: add description of object dictionary
+        """
+        file_list = Path(input_path).glob("all_positions_*")
+        all_pos_list = []
+        for file_path in file_list:
+            with open(file_path, "r") as j:
+                all_pos_list.append(json.loads(j.read()))
+
+        all_objs = {}
+        for packing_id, all_pos in enumerate(all_pos_list):
+            for seed, object_dict in all_pos.items():
+                for obj, positions in object_dict.items():
+                    positions = numpy.array(positions)
+                    if obj not in all_objs:
+                        all_objs[obj] = {}
+                    seed_key = f"{seed}_{packing_id}"
+                    if seed_key not in all_objs[obj]:
+                        all_objs[obj][seed_key] = {}
+                    for ct, dim in enumerate(["x", "y", "z"]):
+                        all_objs[obj][seed_key][dim] = positions[:, ct]
+        self.all_objs = all_objs
+        self.all_pos_list = all_pos_list
+        return all_objs, all_pos_list
+
+    def run_similarity_analysis(self, all_objs):
+        ingr_key = self.ingr_key
+        key_list = list(all_objs[ingr_key].keys())
+        similarity_df = pd.DataFrame(
+            index=key_list,
+            columns=pd.MultiIndex.from_product([["x", "y", "z"], key_list]),
+            dtype=float,
+        )
+        print("Running similarity analysis...")
+        similarity_df["packing_id"] = 0
+
+        for rc, (seed1, pos_dict1) in enumerate(all_objs[ingr_key].items()):
+            similarity_df.loc[seed1, "packing_id"] = seed1.split("_")[-1]
+            for cc, (seed2, pos_dict2) in enumerate(all_objs[ingr_key].items()):
+                for dc, dim in enumerate(["x", "y", "z"]):
+                    arr1 = pos_dict1[dim]
+                    arr2 = pos_dict2[dim]
+                    if len(arr1) == 1 or len(arr2) == 1:
+                        scaled_sig = 0
+                    else:
+                        ad_stat = scipy.stats.anderson_ksamp([arr1, arr2])
+                        scaled_sig = (ad_stat.significance_level - 0.001) / (
+                            0.25 - 0.001
+                        )
+                    similarity_df.loc[seed1, (dim, seed2)] = scaled_sig
+
+        df_packing = similarity_df["packing_id"]
+        lut = dict(zip(df_packing.unique(), sns.color_palette()))
+        row_colors = df_packing.map(lut)
+
+        for dim in ["x", "y", "z"]:
+            g = sns.clustermap(
+                similarity_df[dim],
+                row_colors=row_colors,
+                dendrogram_ratio=(0.15, 0.16),
+                cbar_kws={"label": "similarity score"},
+            )
+            g.ax_heatmap.set_xlabel(f"{ingr_key}_{dim}")
+            g.savefig(f"{self.output_path}clustermap_{ingr_key}_{dim}", dpi=300)
```

Comment:
A print statement where of where you've saved things could be nice for debugging

Reply:
This can get messy while analyzing a large number of packings. I can print `self.output_path` once before starting the workflow.

Reply:
Output path is now printed at the initialization of the analysis workflow